File: src/main/java/me/blake/pulsar/actions/impl/MessageAction.java
package me.blake.pulsar.actions.impl;

import lombok.AllArgsConstructor;
import me.blake.pulsar.actions.Action;
import me.blake.pulsar.messaging.Message;
import org.bukkit.entity.Player;

@AllArgsConstructor
public class MessageAction extends Action {

    private String message;

    public static Action from(String rawAction) {
        return new MessageAction(rawAction);
    }

    @Override
    public void execute(Player player, Message.Placeholder[] placeholders) {
        new Message(message).send(player, placeholders);
    }

    @Override
    public void execute(Player player) {
        new Message(message).send(player);
    }
}

File: src/main/kotlin/me/blake/pulsar/actions/impl/MessageAction.kt
package me.blake.pulsar.actions.impl

import me.blake.pulsar.actions.Action
import me.blake.pulsar.messaging.Message
import me.blake.pulsar.messaging.Message.Placeholder
import org.bukkit.entity.Player

class MessageAction(
    private val message: String,
): Action() {
    override fun execute(player: Player, vararg placeholders: Placeholder) {
        Message(message).send(player, *placeholders)
    }

    override fun execute(player: Player) {
        Message(message).send(player)
    }
}